rocky/rocky/views/tasks.py

```diff
@@ -115,3 +118,88 @@ def get_context_data(self, **kwargs):
         ]
 
         return context
+
+
+class AllTaskListView(ListView):
+    paginate_by = 100
+    paginator_class = RockyPaginator
+    schedulers: list[str] = []
+
+    def get_queryset(self):
+        task_type = self.request.GET.get("type", self.plugin_type)
+        status = self.request.GET.get("scan_history_status", "") or None
+        input_ooi = self.request.GET.get("scan_history_search", "") or None
+        self.schedulers = [f"{task_type}-{o.code}" for o in self.request.user.organizations]
+
+        if self.request.GET.get("scan_history_from"):
+            min_created_at = datetime.strptime(self.request.GET.get("scan_history_from"), "%Y-%m-%d")
+        else:
+            min_created_at = None
+        if self.request.GET.get("scan_history_to"):
+            max_created_at = datetime.strptime(self.request.GET.get("scan_history_to"), "%Y-%m-%d")
+        else:
+            max_created_at = None
+
+        try:
+            return client.get_lazy_task_list(
+                scheduler_id=None,
+                task_type=task_type,
+                status=status,
+                min_created_at=min_created_at,
+                max_created_at=max_created_at,
+                input_ooi=input_ooi,
+                filters={"filters": [{"column": "scheduler_id", "operator": "in", "value": self.schedulers}]},
+            )
+        except HTTPError:
+            error_message = _("Fetching tasks failed: no connection with scheduler")
+            messages.add_message(self.request, messages.ERROR, error_message)
+            return []
+
+    def post(self, request, *args, **kwargs):
+        self.handle_page_action(request.POST["action"])
+        return redirect(request.path)
+
+    def handle_page_action(self, action: str) -> None:
+        if action == PageActions.RESCHEDULE_TASK.value:
+            task_id = self.request.POST.get("task_id")
+            organization_code = self.request.POST.get("organization_code")
```

Comment:
We need to validate that the user can access this organization. There is some code further down the line that checks if the given organization ID matches the Job's organization ID, but checking for access rights needs to be done here.

Reply:
Another route could be to just link to the organization's taskview page action, which because it loads with the organization in the url, will aready do this check for us. That would mean we just need to point the rerun form to that url in the same way the organiaton specific view does it. In that case this whole post handler and handle_page_action ca be removed.

Reply:
You mean something like
```python
organization_code in [o.code for o in self.request.user.organizations]
```
?